sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/CatalogFileIndex.scala

```diff
@@ -84,8 +84,8 @@ class CatalogFileIndex(
       new PrunedInMemoryFileIndex(
         sparkSession, new Path(baseLocation.get), fileStatusCache, partitionSpec, Option(timeNs))
     } else {
-      new InMemoryFileIndex(
-        sparkSession, rootPaths, table.storage.properties, userSpecifiedSchema = None)
+      new InMemoryFileIndex(sparkSession, rootPaths, table.storage.properties,
+        userSpecifiedSchema = None, fileStatusCache)
```

Comment:
`fileStatusCache = fileStatusCache`, since it's already after a named parameter

Reply:
updated